File: app/src/main/java/ga/forntoh/bableschool/model/Score.java
package ga.forntoh.bableschool.model;

@SuppressWarnings("unused")
public class Score {

    private double firstSequenceMark, secondSequenceMark;
    private Course course;
    private String rank;
    private String termRank;

    public double getFirstSequenceMark() {
        return firstSequenceMark;
    }

    public double getSecondSequenceMark() {
        return secondSequenceMark;
    }

    public Course getCourse() {
        return course;
    }

    public String getRank() {
        return rank;
    }

    public String getTermRank() {
        return termRank;
    }

    public double getScoreAverage() {
        if (firstSequenceMark < 0 || secondSequenceMark < 0) return -1;
        return (firstSequenceMark + secondSequenceMark) / 2.0;
    }
}

File: app/src/main/java/ga/forntoh/bableschool/data/model/Score.kt
package ga.forntoh.bableschool.data.model

import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.ForeignKey
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table
import ga.forntoh.bableschool.data.db.AppDatabase
import ga.forntoh.bableschool.data.model.main.Course

@Table(database = AppDatabase::class)
data class Score(
        @PrimaryKey(autoincrement = true) var id: Long = 0,
        @ForeignKey(saveForeignKeyModel = true) var course: Course? = null,
        @Column var firstSequenceMark: Double = 0.toDouble(),
        @Column var secondSequenceMark: Double = 0.toDouble(),
        @Column var rank: String? = null,
        @Column var termRank: String? = null
) {
    val scoreAverage: Double get() = if (firstSequenceMark < 0 || secondSequenceMark < 0) -1.0 else (firstSequenceMark + secondSequenceMark) / 2.0
    @Column
    var term: Int = 1
}